### PR TITLE
TC04 - Contact Email contains correct href value 

### PR DESCRIPTION
**Pull request by sattdn21-01-02**

### CODE REVIEW

File: src/test/java/com/logigear/ContactTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    private final ContactPage contactPage = new ContactPage();
+
+    @Test(description = "TC04 - Contact Email contains correct href value which can help to quickly open Outlook Compose Message dialog")
+    public void TC04() {
+        LoggerHelper.startTestCase("TC04 - Contact Email contains correct href value which can help to quickly open Outlook Compose Message dialog");
+        String contactEmail = "mailto:thanh.viet.le@logigear.com";
+        homePage.goToContactPage();
+
+        String actualMsg = contactPage.getEmail();
+        String expectedMsg = contactEmail;
```
Comment (tanmle) [inline src/test/java/com/logigear/ContactTest.java:24]: hmm, this is really bad, you already have `contactEmail`, why do you still define `exptectMsg`?

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/test/java/com/logigear/ContactTest.java (hunk 1 above)
Comment (sattdn21-01-02) [reply src/test/java/com/logigear/ContactTest.java:24]: Thanks for your review. I just have fixed and pushed into this branch. Please review my newest update.

File: src/main/java/page_objects/ContactPage.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import helper.BrowserHelper;
+import helper.ElementHelper;
+import helper.element_helper.Link;
+import org.openqa.selenium.By;
+
+public class ContactPage extends GeneralPage {
+
+    //Elements
+    private final By linkEmail = new By.ByLinkText("thanh.viet.le@logigear.com");
```
Comment (tanmle) [inline src/main/java/page_objects/ContactPage.java:11]: nooooooooooooooooooooooooooo, why do you capture an element like this?

File: src/test/java/com/logigear/ContactTest.java (hunk 1 above)
Comment (tanmle) [reply src/test/java/com/logigear/ContactTest.java:24]: I don't see any update on this.

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/main/java/page_objects/ContactPage.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    //Methods
+    public String getEmail() {
```
Comment (fey031) [inline src/main/java/page_objects/ContactPage.java:14]: getEmailHref()

**Review by fey031 — CHANGES_REQUESTED** [review]

**Review by tanmle — APPROVED** [review]

File: src/test/java/com/logigear/ContactTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    private final ContactPage contactPage = new ContactPage();
+
+    @Test(description = "TC04 - Contact Email contains correct href value which can help to quickly open Outlook Compose Message dialog")
+    public void TC04() {
+        LoggerHelper.startTestCase("TC04 - Contact Email contains correct href value which can help to quickly open Outlook Compose Message dialog");
+        String contactEmail = "mailto:thanh.viet.le@logigear.com";
+        homePage.goToContactPage();
+
+        String actualMsg = contactPage.getEmailHref();
+        Assert.assertEquals(actualMsg, contactEmail);
```
Comment (fey031) [inline src/test/java/com/logigear/ContactTest.java:24]: Add assert message here

File: src/test/java/com/logigear/ContactTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    private final HomePage homePage = new HomePage();
+    private final ContactPage contactPage = new ContactPage();
+
+    @Test(description = "TC04 - Contact Email contains correct href value which can help to quickly open Outlook Compose Message dialog")
+    public void TC04() {
+        LoggerHelper.startTestCase("TC04 - Contact Email contains correct href value which can help to quickly open Outlook Compose Message dialog");
+        String contactEmail = "mailto:thanh.viet.le@logigear.com";
+        homePage.goToContactPage();
+
+        String actualMsg = contactPage.getEmailHref();
```
Comment (fey031) [inline src/test/java/com/logigear/ContactTest.java:23]: actualMsg --> actualEmailHref

File: src/test/java/com/logigear/ContactTest.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        Assert.assertEquals(actualMsg, contactEmail,"Error message fails to display!");
```
Comment (fey031) [inline src/test/java/com/logigear/ContactTest.java:24]: insert space character
Assert.assertEquals(actualMsg, contactEmail, "{} href should be appeared".format(contactEmail));

File: src/test/java/com/logigear/LoginTest.java [hunk 7]
```diff
@@ -26,6 +25,6 @@ public void TC01() {
 
         String actualMsg = homePage.getWelcomeMessage();
         String expectedMsg = Constant.WELCOME + Constant.USERNAME;
-        Assert.assertEquals(actualMsg, expectedMsg);
+        Assert.assertEquals(actualMsg, expectedMsg,"Welcome message fails to display!");
```
Comment (fey031) [inline src/test/java/com/logigear/LoginTest.java:28]: insert space

**Review by fey031 — CHANGES_REQUESTED** [review]

**Review by fey031 — APPROVED** [review]
LGTM

**Pull request merged**